time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateTimeArbitrary.java

```diff
@@ -21,14 +21,21 @@
 	private LocalDateTime min = null;
 	private LocalDateTime max = null;
 
+	private LocalDate minDate = null;
```

Comment:
There's so much duplication with LocalDate and LocalTime arbitrary implementations. I wonder if there's a way to get rid of that.

Reply:
I don't think so because you need it for the calculation of effectiveMin/effectiveMax which is different from the calculation in the time/date class.

Some things (i.e. monthMin/Max) you can exclude because you don't need them for the calculation but then it ist not consistent. Should I exclude these things?

Reply:
I’m not saying simple delegation is possible but there should be a way to put all the duplicated logic of checking validity and creating error messages/exceptions in a common (new) abstraction.

Reply:
To make sure that I understand you right:
Do you mean a class like the new class LocalDateTimeSupport?

Do you have a better name for it?
In which package is the class correct?

Reply:
I was thinking of methods like `yearBetween()` and `monthBetween()` etc. They should be the same (structurally and as of error/special case logic). But if you compare `DefaultLocalDateArbitrary.yearBetween()` with `DefaultLocalDateTimeArbitrary.yearBetween()` you will see that they behave differently. Unifying both in s.t. like a `YearBetween` class could do two things: 1) Get rid of duplicated code 2) Show where the logic is different although it should be the same.

Reply:
> Unifying both in s.t. like a `YearBetween` class could do two things: 1) Get rid of duplicated code 2) Show where the logic is different although it should be the same.

I'm not sure this is the best design approach but in my head there's a small hierarchy looming

```
ValueRange<T, U>
    YearBetween<Integer, LocalDate>
    MonthBetween<Integer, LocalDate>
    DayOfMonthBetween<Integer, Integer>
...
```

That's just a sketch, of course.

Reply:
I think it is not completly like you expected but it works.